### PR TITLE
Adds docs for GameEngine

### PR DESCRIPTION
**Pull request by pathunstrom**

### CODE REVIEW

File: docs/reference/engine.rst [hunk 1]
```diff
@@ -0,0 +1,16 @@
+.. py:currentmodule:: ppb.engine
+
+================
+GameEngine
+================
+
+The ``GameEngine`` is the literal beating heart of ppb: It's the publisher for
```
Comment (astronouth7303) [inline docs/reference/engine.rst:7]: "publisher" feels like the wrong word.

Comment (astronouth7303) [inline docs/reference/engine.rst:7]:
```suggestion
The :py:class:`GameEngine` is the literal beating heart of ppb: It's the publisher for
```

File: docs/reference/engine.rst [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+the event and messaging system, is the source of the ``Idle`` event, and is the
+root container of the object tree.
+
+Some of the engine of the API is definitely intended for advanced users. Use
+the various methods of GameEngine with caution.
```
Comment (astronouth7303) [inline docs/reference/engine.rst:12]:
```suggestion
the various methods of :py:class:`GameEngine` with caution.
```

File: docs/reference/engine.rst (hunk 1 above)
Comment (pathunstrom) [reply docs/reference/engine.rst]: Better?

File: ppb/engine.py [hunk 3]
```diff
@@ -262,6 +333,14 @@ def flush_events(self):
         self.events = deque()
 
     def walk(self):
+        """
+        Walk the object tree.
+
+        Publication order: The :class:`GameEngine`, the :class:`GameEngine`
```
Comment (astronouth7303) [inline ppb/engine.py:339]: You list GameEngine twice

Comment (pathunstrom) [reply ppb/engine.py:339]: That's "The GameEngine System list". It probably reads fine without it, though.

**Pull request merged**